lib/ui/auth/login.dart

```diff
@@ -1,106 +1,120 @@
   final String title;
+
   LoginPage({Key key, this.title}) : super(key: key);
 
-  void _handleLogin() {
-    //Handle login
-  }
+  @override
+  _LoginPageState createState() => _LoginPageState();
+}
 
-  void _handleGoogleLogin() {
-    //Handle login
+class _LoginPageState extends State<LoginPage> {
+  @override
+  void initState() {
+    super.initState();
+    final vm = Provider.of<LoginViewModel>(context, listen: false);
+    vm.resetState();
   }
 
   void _handleCreateAccount(BuildContext context) {
     Navigator.push(
       context,
-      MaterialPageRoute(builder: (context) => SignUpPage(title: title)),
+      MaterialPageRoute(builder: (context) => SignUpPage(title: widget.title)),
     );
   }
 
-  @override
-  Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        // Here we take the value from the MyHomePage object that was created by
-        // the App.build method, and use it to set our appbar title.
-        title: Text(title),
-      ),
-      body: Padding(
-        padding: EdgeInsets.all(24.0),
-        child: Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Text(
-              'Log In',
-              textAlign: TextAlign.start,
-              style: Theme.of(context).textTheme.headline5,
-            ),
-            Row(
-              children: <Widget>[
-                Text(
-                  "Don\'t have an account? ",
-                  textAlign: TextAlign.start,
-                  style: TextStyle(color: Colors.grey.shade100),
-                ),
-                TextButton(
-                  style: ButtonStyle(
-                    padding: MaterialStateProperty.all(
-                        EdgeInsets.symmetric(vertical: 0, horizontal: 2)),
-                  ),
-                  onPressed: () { _handleCreateAccount(context); },
-                  child: Text(
-                    "Create one here.",
-                    textAlign: TextAlign.start,
-                    style: TextStyle(fontWeight: FontWeight.bold),
-                  ),
-                ),
-              ],
-            ),
-            TextField(
-              obscureText: false,
-              decoration: InputDecoration(
-                labelStyle: TextStyle(color: Colors.grey.shade100),
-                border: OutlineInputBorder(),
-                labelText: 'Email',
-              ),
-            ),
-            SizedBox(height: 16),
-            TextField(
-              obscureText: true,
-              decoration: InputDecoration(
-                labelStyle: TextStyle(color: Colors.grey.shade100),
-                border: OutlineInputBorder(),
-                labelText: 'Password',
+  Widget _authScreen(BuildContext context, LoginViewModel viewModel) {
+    if (viewModel.state.authStatus == AuthStatus.ERROR) {
+      Future.delayed(Duration.zero, () async {
```

Comment:
Is this a hack to get the AuthDialog to show? Seems like a delay of zero wouldn't have a huge benefit

Reply:
Yeah, this is a weird workaround that I found. I'm not sure why, but If I present the dialog before returning a widget I get an error. This allows for the widget to be computed before showing the dialog. There is probably a better way to do this though 🤷 
![unnamed](https://user-images.githubusercontent.com/16354865/112877251-27a6dc00-908c-11eb-8df7-6cf3f3c30079.jpg)